std-bits/table/src/main/java/org/enso/table/data/table/join/JoinStrategy.java

```diff
@@ -33,9 +34,9 @@ static JoinStrategy createStrategy(List<JoinCondition> conditions, JoinKind join
       assert !betweenConditions.isEmpty();
       return new SortJoin(betweenConditions, builderSettings);
     } else if (betweenConditions.isEmpty()) {
-      return new HashJoin(hashableConditions, new MatchAllStrategy(), builderSettings);
+      return new SimpleHashJoin(hashableConditions, new MatchAllStrategy(), builderSettings);
     } else {
-      return new HashJoin(
+      return new CompoundHashJoin(
           hashableConditions, new SortJoin(betweenConditions, builderSettings), builderSettings);
```

Comment:
I feel like a have a de-ja-vu - I think this is similar to what this was before one of my refactors 😅 when I was trying to limit duplicated logic.

But I think that with additional optimizations it indeed makes sense to split back.